docs/guide/snaps-development-guide.md

```diff
@@ -1,47 +1,290 @@
 
-In addition, you can review the growing number of [example snaps](https://github.com/MetaMask/snaps-skunkworks/tree/main/packages/snap-examples) maintained by MetaMask.
+1. Extend the dapp-facing MetaMask JSON-RPC API in arbitrary ways.
+2. Integrate with and extend the functionality of MetaMask using the [snaps RPC methods and permissions](./snaps-rpc-api.html).
 
-Finally, you may wish to check out the first functional key management snap ever made, [the Filecoin Snap, or `filsnap`](https://github.com/Chainsafe/filsnap/).
+In this section, we'll go into detail about how to actually develop a snap and overcome common issues encountered during development.
 
-## Gotchas
+### The Snap Lifecycle
 
-### Accessing the Internet in a Snap
+Before beginning the development of your snap, it's important to understand the snap lifecycle.
+Just like [service workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) or AWS lambda functions, snaps are designed to wake up in response to messages / events, and shut down when they are idle.
+We say that snaps have an "ephemeral" lifecycle; here one moment, gone the next.
+In addition, if MetaMask detects that a snap becomes unresponsive, it will be shut down.
+This does not mean that you can't create long-running snaps, but it does mean that your snaps must handle being shut down, especially when they are not within the JSON-RPC request / response cycle.
+
+A snap is considered "unresponsive" if:
+
+1. It has not received a JSON-RPC request for 30 seconds.
+2. It takes more than 60 seconds to process a JSON-RPC request.
+
+Stopped snaps are started whenever they receive a JSON-RPC request, unless they have been disabled.
+If a snap is disabled, the user must re-enable it before it can start again.
+
+### Permissions
+
+Just like dapps need to request the `eth_accounts` permission in order to access the user's Ethereum accounts, so do snaps need to request access to the sensitive methods in the snaps RPC API.
+Snaps can effectively expand the MetaMask RPC API by implementing their own using `wallet.registerRpcMessageHandler()`, but in order integrate deeply with MetaMask, you need to make use of the Snaps RPC API's [restricted methods](./snaps-rpc-api.html#restricted-methods).
+Access restriction is implemented using [EIP-2255 wallet permissions](https://eips.ethereum.org/EIPS/eip-2255), and you must specify the permissions required by your snap in the manifest's `initialPermissions` field.
+You can find an example of how to do this in the [template snap's manifest](https://github.com/MetaMask/template-snap/blob/main/snap.manifest.json).
+
+#### Accessing the Internet
 
 Snaps do not get access to any sensitive APIs or features by default, and Internet access is no exception to that rule.
 To access the Internet, you must be specify the permission `endowment:network-access` in the `initialPermissions` of your `snap.manifest.json` file.
```

Comment:
Should be: "To access the internet, you must specify the permission `endowment:network-access` in the `initialPermissions` of your `snap.manifest.json` file.

(remove 'be')